spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/ReactorSleuthMethodInvocationProcessor.java

```diff
@@ -146,7 +146,9 @@ public void subscribe(CoreSubscriber<? super Object> actual) {
 			Span span;
 			Tracer tracer = this.processor.tracer();
 			if (this.span == null) {
-				span = tracer.newTrace();
+				// If we aren't continuing a trace from this flow, use nextSpan so that it
+				// can consider the "current span" (typically, backed by a thread-local)
+				span = tracer.nextSpan();
```

Comment:
Many tests in https://github.com/spring-cloud/spring-cloud-sleuth/blob/d789454147e84ece9adfaea4d994d38834881d5d/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/ReactorSleuthMethodInvocationProcessor.java hit this line, except all result in a new root trace.